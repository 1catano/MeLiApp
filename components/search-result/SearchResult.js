import React from "react";
import PropTypes from "prop-types";
import ProductCard from "../product-card/ProductCard";
import { useRouter } from "next/router";
import sty from "./search-result.module.scss";

const SearchResult = (props) => {
    const { items } = props;
    const router = useRouter();
    const handleRedirect = (id) => {
        router.push(`items/${id}`);
    };
    return (
        <section className={sty["ml-search-result__main"]}>
            {items.map((item, idx) => {
                return (
                    <ProductCard
                        productId={item.id}
                        price={item.price}
                        title={item.title}
                        picture={item.picture}
                        showDivider={items.length - 1 > idx}
                        stateName={item.state_name}
                        isFreeShipping={item.free_shipping}
                        handleRedirect={handleRedirect}
                    />
                );
            })}
        </section>
    );
};

SearchResult.propTypes = {
    items: PropTypes.arrayOf(
        PropTypes.shape({
            id: PropTypes.string,
            price: PropTypes.shape({
                currency: PropTypes.string,
                amount: PropTypes.number,
                decimals: PropTypes.number,
            }),
            title: PropTypes.string,
            picture: PropTypes.string,
            state_name: PropTypes.string,
            free_shipping: PropTypes.bool,
        }),
    ),
};

SearchResult.defaultProps = {
    items: [],
};

export default SearchResult;
