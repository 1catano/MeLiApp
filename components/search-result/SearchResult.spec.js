import React from "react";
import { shallow } from "enzyme";
import SearchResult from "./SearchResult";
describe("SearchResult snapshot", () => {
    it("take snapshot", () => {
        const component = shallow(<SearchResult />);
        expect(component).toMatchSnapshot();
    });
});
