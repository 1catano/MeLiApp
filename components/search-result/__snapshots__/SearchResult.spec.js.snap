// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResult snapshot render amount and divider 1`] = `ShallowWrapper {}`;

exports[`SearchResult snapshot take snapshot 1`] = `ShallowWrapper {}`;
